engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java

```diff
@@ -76,13 +85,26 @@ public void initContext() {
   }
 
   @After
-  public void disposeContext() {
+  public void disposeContext() throws IOException {
     context.close();
     context = null;
     engine.close();
     engine = null;
   }
 
+  /** Only print warnings from the compiler if a test fails. */
+  @Rule
+  public TestWatcher testWatcher =
+      new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
```

Comment:
Interesting. If this could be the default for `defaultContextBuilder`.

Reply:
I was thinking the same way. But I am afraid that in order to implement this, [ContextUtils](https://github.com/enso-org/enso/blob/fb821303ab2818c4ce2d8482612584497f2e2014/lib/java/test-utils/src/main/java/org/enso/test/utils/ContextUtils.java) would need to be an abstract class and all the test classes would need to extend it.

Reply:
Tracked in https://github.com/enso-org/enso/issues/11683